View a list of all reviews

PeterM24 opened the pull request:
### List of review cards:
![IMG_4486](https://user-images.githubusercontent.com/123666879/232781054-7b2d3a6a-b29d-4058-b201-b45836d0e3e3.png)

### Loading:
![Review loading](https://user-images.githubusercontent.com/123666879/232780969-ff4c77e2-b338-4895-8d05-ecc109e67bbf.gif)

PeterM24 committed:
add background image

PeterM24 committed:
update api and app

PeterM24 committed:
update components

PeterM24 committed:
update interfaces

PeterM24 committed:
update css

PeterM24 committed:
plan

aloush-dev commented:
Can you please resubmit this in JavaScript and not TypeScript.
Thanks

NRMSMITH commented:
-make sure you have some loading functionality :) 
-put api.tsx in a utils folder just to keep the root nice and neat

PeterM24 committed:
add loading feature to list reviews

PeterM24 committed:
refactor backgrounds for accessibility - media queries for devices and variable image sizes

PeterM24 commented:
> -make sure you have some loading functionality :)
> -put api.tsx in a utils folder just to keep the root nice and neat

Thanks @NRMSMITH - I've added loading functionality and added a sample gif / img to the PR if you want to have a look.

Thanks :)

PeterM24 committed:
Merge branch 'main' into list-all-reviews

popatre commented on src/App.tsx:
```diff
@@ -1,10 +1,23 @@
+        setIsLoading={setIsLoading}
+        reviewList={reviewList}
+        setReviewList={setReviewList}
+      />
```
I'd ideally start setting up your routing i.e. what url will load this list of reviews component?

popatre commented on src/App.tsx:
```diff
@@ -1,10 +1,23 @@
+import Background from "./components/Background";
+
+function App(): JSX.Element {
+  const [reviewList, setReviewList] = useState<Review[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```
State wise -this could live lower, im not sure what else would use the loading state of the ListOfReviews component. I dont think anything else will use the reviewList either. 

ideally state needs to be as low as possible.

popatre commented on src/components/ListReviews.tsx:
```diff
@@ -0,0 +1,42 @@
+  return (
+    <div>
+      <h2 className="page-title">List of Reviews</h2>
+      {/* {isLoading ? <LoadingReviewCard/> : null} */}
```
remove commented out code

popatre commented on src/components/ReviewCard.tsx:
```diff
@@ -0,0 +1,24 @@
+import { Review } from "../interfaces/review.interface";
+
+const ReviewCard = ({ review }: { review: Review }): JSX.Element => {
+  const date = new Date(review.created_at).toLocaleString();
+
+  return (
+    <li key={review.review_id} className="list-item">
+      <h3 className="list-item-header">{review.title}</h3>
+      <h5 className="list-item-designer">{review.designer}</h5>
+      <h5 className="list-item-owner">{review.owner}</h5>
+      <img
+        src={review.review_img_url}
+        alt={`image for ${review.title}`}
+        className="list-item-image"
+      />
+      <p className="list-item-body">
+        {review.review_body.slice(1, 300)} ... read more
+      </p>
+      <p className="list-item-date">Date posted: {date}</p>
+    </li>
```
Nice display component here

popatre reviewed (changes requested):
Could you start to set up your routes ie. a home `/` for this one. 
Put you in a better place for the next tasks. 

also, i'd move the state out the app. Only put it up there if needs be. I dont think anything else will use that state, but you can always move it if so

PeterM24 committed:
move useState into specific components

PeterM24 committed:
move useState into specific components
